scripts/private/_send-packet-config.js

```diff
@@ -8,43 +8,46 @@ if (!source) {
   process.exit(1);
 }
 
-function runSendPacket(config) {
+function runSendPacketCommand(command) {
+  console.log("Running sendPacket command: ", command);
+  return new Promise((resolve, reject) => {
+    exec(command, (error, stdout, stderr) => {
+      if (error) {
+        console.error(`exec error: ${error}`);
+        reject(error);
+      } else {
+        console.log(stdout);
+        resolve(true);
+      }
+    });
+  });
+}
+
+async function runSendPacket(config) {
   // Check if the source chain from user input is whitelisted
   const allowedNetworks = getWhitelistedNetworks();
   if (!allowedNetworks.includes(source)) {
     console.error("❌ Please provide a valid source chain");
     process.exit(1);
   }
 
-  // Run the send-packet or send-universal-packet script based on the config
-  if (config.isUniversal) {
-    exec(`npx hardhat run scripts/send-universal-packet.js --network ${source}`, (error, stdout, stderr) => {
-        if (error) {
-          console.error(`exec error: ${error}`);
-          return;
-        } else {
-          console.log(stdout);
-        }
-    });
-  } else {
-    exec(`npx hardhat run scripts/send-packet.js --network ${source}`, (error, stdout, stderr) => {
-        if (error) {
-          console.error(`exec error: ${error}`);
-          return;
-        }else {
-          console.log(stdout);
-        }
-    });
+  const script = config.isUniversal ? 'send-universal-packet.js' : 'send-packet.js';
+  const command = `npx hardhat run scripts/${script} --network ${source}`;
+
+  try {
+    await setupIbcPacketEventListener();
+    await runSendPacketCommand(command);
+  } catch (error) {
+    console.error("❌ Error sending packet: ", error);
+    process.exit(1);
```

Comment:
it should run listenEvent first, otherwise some event can't be catch.
This PR is to make sure the program stops correctly & log when an error happens.